schemas/assessment.json

```diff
@@ -5,30 +5,43 @@
   "type": "object",
   "additionalProperties": false,
   "required": [
+    "response_status",
+    "error_code",
     "threshold_condition_hash",
+    "reported_symptoms_array",
+    "experiencing_symptoms",
     "patient_submission_token"
   ],
   "properties": {
+    "response_status": {
+      "description": "Twilio response status information and codes, if present",
+      "enum": [ "opt_out", "opt_in", "no_answer_voice", "no_answer_sms", "error_voice", "error_sms", "max_retries_sms", "max_retries_voice", "success_sms", "success_voice", null]
+    },
+    "error_code": {
+      "description": "A Twilio error code",
+      "type": ["null", "string"]
+    },
     "threshold_condition_hash": {
-      "type": "string",
       "description": "A way for an assessment to reference the set of symptoms and expected values that are associated with the assessment",
+      "type": ["string", "null"],
       "pattern": "[a-f0-9]{64}"
     },
     "patient_submission_token": {
-      "type": "string",
-      "description": "Unique token used to validate existence of a patient for this assessement's submission.",
-      "pattern": "[a-f0-9]{40}"
+      "description": "Unique token used to validate existence of a patient for this assessement's submission. Supports old 40 character token, new 10 character tokens, and 34 character execution flow IDs",
+      "type": ["string", "null"],
+      "anyOf": [
+        { "pattern": "^[a-f0-9]{40}$" },
+        { "pattern": "^[a-zA-Z0-9_-]{10}$" },
+        { "pattern": "^FN[a-f0-9]{32}$" }
+      ]
     },
     "experiencing_symptoms": {
       "description": "Represents an assessment response over voice.",
-      "anyOf": [
-        { "type": "boolean" },
-        { "type": "null" }
-      ]
+      "type": ["boolean", "null"]
     },
     "reported_symptoms_array": {
```

Comment:
Can't comment on the actual line, but in the `reported_symptoms_array`, I think we need to add a property `required` to the list of required properties. So
```json
        "required": [
          "name",
          "value",
          "type",
          "label",
          "notes",
          "required"
        ],
```
Since from what I'm seeing, `required` is a boolean we set on the symptoms array [here](https://github.com/SaraAlert/SaraAlert/blob/9cdc2fe10ede7ee09f5701f779b57203b5468aeb/app/controllers/assessments_controller.rb#L135). So for me the validation was failing because this additional disallowed property was present.